src/tools/x/src/main.rs

```diff
@@ -0,0 +1,32 @@
+use std::{
+    env,
+    os::unix::process::CommandExt,
+    process::{self, Command},
+};
+
+fn main() {
+    let current = match env::current_dir() {
+        Ok(dir) => dir,
+        Err(err) => {
+            eprintln!("Failed to get current directory: {}", err);
+            process::exit(1);
+        }
+    };
+
+    for dir in current.ancestors() {
+        let candidate = dir.join("x.py");
+
+        if candidate.exists() {
+            let error = Command::new(&candidate).args(env::args().skip(1)).current_dir(dir).exec();
```

Comment:
While you're being clever, `./x.py` doesn't work on Windows - can you change this to run `python x.py` directly?

Reply:
And actually, if you could look up `python`, `python3`, `python2` and look up the first one that works that would be _amazing_. https://github.com/rust-lang/rust/issues/71818

Reply:
Done!